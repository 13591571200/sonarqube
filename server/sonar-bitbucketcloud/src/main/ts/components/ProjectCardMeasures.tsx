/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { Language } from '@sqcore/api/languages';
import MeasureBugs from './measures/MeasureBugs';
import MeasureCodeSmells from './measures/MeasureCodeSmells';
import MeasureCoverage from './measures/MeasureCoverage';
import MeasureDuplication from './measures/MeasureDuplication';
import MeasureLanguages from './measures/MeasureLanguages';
import MeasureVulnerabilities from './measures/MeasureVulnerabilities';
import { ProjectData } from '../types';

interface Props extends Pick<ProjectData, 'measures'> {
  languages: Language[];
}

export default function ProjectCardMeasures({ languages, measures }: Props) {
  return (
    <div className="project-card-measures">
      <div className="project-card-measures-left">
        <MeasureBugs measures={measures} />
        <i className="project-card-measure-separator" />
        <MeasureVulnerabilities measures={measures} />
        <i className="project-card-measure-separator" />
        <MeasureCodeSmells measures={measures} />
        <i className="project-card-measure-separator hidden-small" />
      </div>
      <div className="project-card-measures-right">
        <MeasureCoverage measures={measures} />
        <i className="project-card-measure-separator" />
        <MeasureDuplication measures={measures} />
        <i className="project-card-measure-separator hidden-big" />
        <MeasureLanguages languages={languages} measures={measures} />
      </div>
    </div>
  );
}
