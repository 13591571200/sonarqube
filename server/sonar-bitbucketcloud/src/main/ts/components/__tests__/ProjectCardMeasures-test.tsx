/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable camelcase */
import * as React from 'react';
import { shallow } from 'enzyme';
import ProjectCardMeasures from '../ProjectCardMeasures';

const MEASURES = {
  alert_status: 'ERROR',
  bugs: '17',
  code_smells: '132',
  coverage: '88.3',
  duplicated_lines_density: '9.8',
  ncloc: '2053',
  ncloc_language_distribution: 'js=2004;java=564644',
  reliability_rating: '1.0',
  security_rating: '1.0',
  sqale_rating: '1.0',
  vulnerabilities: '0'
};

it('should display correctly', () => {
  expect(shallow(<ProjectCardMeasures languages={[]} measures={MEASURES} />)).toMatchSnapshot();
});
