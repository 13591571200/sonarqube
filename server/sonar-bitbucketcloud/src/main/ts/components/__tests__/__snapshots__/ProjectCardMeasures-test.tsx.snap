// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly 1`] = `
<div
  className="project-card-measures"
>
  <div
    className="project-card-measures-left"
  >
    <MeasureBugs
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
    <i
      className="project-card-measure-separator"
    />
    <MeasureVulnerabilities
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
    <i
      className="project-card-measure-separator"
    />
    <MeasureCodeSmells
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
    <i
      className="project-card-measure-separator hidden-small"
    />
  </div>
  <div
    className="project-card-measures-right"
  >
    <MeasureCoverage
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
    <i
      className="project-card-measure-separator"
    />
    <MeasureDuplication
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
    <i
      className="project-card-measure-separator hidden-big"
    />
    <MeasureLanguages
      languages={Array []}
      measures={
        Object {
          "alert_status": "ERROR",
          "bugs": "17",
          "code_smells": "132",
          "coverage": "88.3",
          "duplicated_lines_density": "9.8",
          "ncloc": "2053",
          "ncloc_language_distribution": "js=2004;java=564644",
          "reliability_rating": "1.0",
          "security_rating": "1.0",
          "sqale_rating": "1.0",
          "vulnerabilities": "0",
        }
      }
    />
  </div>
</div>
`;
